flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/ChunkSplitter.java

```diff
@@ -224,21 +235,73 @@ private MySqlSnapshotSplit createSnapshotSplit(
     // ------------------------------------------------------------------------------------------
 
     /** Checks whether split column is evenly distributed across its range. */
-    private static boolean splitColumnEvenlyDistributed(Column splitColumn) {
-        // only column is auto-incremental are recognized as evenly distributed.
-        // TODO: we may use MAX,MIN,COUNT to calculate the distribution in the future.
-        if (splitColumn.isAutoIncremented()) {
-            DataType flinkType = MySqlTypeUtils.fromDbzColumn(splitColumn);
-            LogicalTypeRoot typeRoot = flinkType.getLogicalType().getTypeRoot();
-            // currently, we only support split column with type BIGINT, INT, DECIMAL
-            return typeRoot == LogicalTypeRoot.BIGINT
-                    || typeRoot == LogicalTypeRoot.INTEGER
-                    || typeRoot == LogicalTypeRoot.DECIMAL;
+    private static boolean splitColumnEvenlyDistributed(
+            TableId tableId, Column splitColumn, Object min, Object max, long approximateRowCnt) {
+        DataType flinkType = MySqlTypeUtils.fromDbzColumn(splitColumn);
+        LogicalTypeRoot typeRoot = flinkType.getLogicalType().getTypeRoot();
+
+        // currently, we only support the optimization that split column with type BIGINT, INT,
+        // DECIMAL
+        final boolean isNumericColumn =
+                typeRoot == LogicalTypeRoot.BIGINT
+                        || typeRoot == LogicalTypeRoot.INTEGER
+                        || typeRoot == LogicalTypeRoot.DECIMAL;
+
+        // only column is numeric and evenly distribution factor is less than
+        // MAX_EVENLY_DISTRIBUTION_FACTOR
+        // will be treated as evenly distributed.
+        if (isNumericColumn) {
```

Comment:
I think we can have 2 more optimizations here:

(1) return true if `max - min <= splitSize` 
(2) move `queryApproximateRowCnt` after (1) to avoid unnecessary computation.